### PR TITLE
feat: refactor text extractor using ledongthuc/pdf & unidoc/unioffice

### PR DESCRIPTION
**Pull request by kv-chiu**

### CODE REVIEW

**Comment by kv-chiu** [comment]
todo:
- adapt to embedding

Please don't merge for now.

File: extractors/util.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	hash.Write([]byte(originalFileName))
+	hashValue := hex.EncodeToString(hash.Sum(nil))
+
+	cacheFileName := fmt.Sprintf("%d_%s.cache", timestamp, hashValue)
+
+	return cacheFileName
+}
+
+func getLocalPathFromUrl(url string) (string, error) {
+	response, err := http.Get(url)
```
Comment (hsluoyz) [inline extractors/util.go:27]: Your code is too complex. Refer to:

```go
func GetUploadXlsxPath(fileId string) string {
	file, err := ioutil.TempFile("", fileId)
	if err != nil {
		panic(err)
	}

	return file.Name()
}
```

File: extractors/pdf_text.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	}
+
+	mergedText := strings.Join(mergedTexts, "\n")
+
+	fmt.Println(mergedText)
+
+	return mergedText, nil
+}
+
+func isSameLine(text1, text2 pdf.Text) bool {
```
Comment (hsluoyz) [inline extractors/pdf_text.go:74]: Don't define this

File: ai/embedding.go [hunk 3]
```diff
@@ -17,25 +17,38 @@ package ai
...
-		return "", err
+func ReadFileToString(fileUrl, fileName string) (string, error) {
+	suffix := strings.ToLower(fileName[strings.LastIndex(fileName, "."):])
+
+	var res string
+	var err error
+
+	switch suffix {
+	case ".pdf":
+		res, err = extractor.GetPdfTextFromUrl(fileUrl)
```
Comment (hsluoyz) [inline ai/embedding.go:36]: First download file contents

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Pull request merged**